src/rollup/types.d.ts

```diff
@@ -203,6 +205,7 @@ export interface Plugin {
 	footer?: AddonHook;
 	intro?: AddonHook;
 	outro?: AddonHook;
+	watchChange?: (id: string) => void;
```

Comment:
So the idea is to deprecate asset dependencies in favour of the `watchChange` handler? Sounds really good to me, much clearer and easier to reason about.

In order to start the actual deprecation soon, should we
* Remove the "dependencies" version `emitAssets` from the types, or at least
* Add an [@deprecated](http://usejsdoc.org/tags-deprecated.html) JSDoc comment above the type which should be picked up by many IDEs? (would be an interesting thing to do for other deprecations as well. Not 100% sure about VSCode but I know WebStorm/IntelliJ uses these `@deprecated` tags to visually mark usages as deprecated).

Also I agree with @shellscape that a warning would make sense considering asset dependencies are not widely known yet anyway (thanks @tivac for pointing that out 😉)